### PR TITLE
fix(starlette/fastapi): add back `aggregate_resources` feature

### PR DESCRIPTION
**Pull request by ZStriker19**
## Description
`aggregate_resources` was removed by https://github.com/DataDog/dd-trace-py/pull/3612 which breaks our public interface contract.

## Checklist
- [ ] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Testing strategy
Added tests to Fastapi and Starlette as well as recorrected the test that was supposed to test this feature.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

**Comment by P403n1x87** [comment]
I think we would want to merge #3850 first and then use `DDTraceDeprecationWarning` with `deprecated`

**Review by brettlangdon — COMMENTED** [review]
+1 agree with Gab re `DDTraceDeprecatonWarning`, otherwise lgtm

File: ddtrace/contrib/starlette/patch.py [hunk 1]
```diff
@@ -93,6 +94,14 @@ def unpatch():
 
 
 def traced_handler(wrapped, instance, args, kwargs):
+    if config.starlette.get("aggregate_resources") is False or config.fastapi.get("aggregate_resources") is False:
+        deprecate(
+            "ddtrace.contrib.starlette.patch",
+            message="`aggregate_resources` is deprecated and will be removed in tracer version 2.0.0",
```
Comment (brettlangdon) [inline ddtrace/contrib/starlette/patch.py:100]:
```suggestion
            message="`aggregate_resources` is deprecated and will be removed in tracer version 2.0.0",
            category=DDTraceDeprecationWarning,
```

**Review by mabdinur — APPROVED** [review]
LGTM

**Review by majorgreys — APPROVED** [review]

**Pull request merged**